src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,184 @@
+        user.setLogin("sanya67");
+        user.setAge(18);
+        user.setPassword("123456");
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+                    registrationService.register(user1);
+                });
+    }
+
+    @Test
+    void loginIsNull_NotOk() {
+        user.setLogin(null);
+        user.setId(11l);
+        user.setPassword("890765");
+        user.setAge(20);
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+            registrationService.register(user);
+                });
+    }
+
+    @Test
+    void loginNotExist_Ok() {
+        user.setId(8l);
+        user.setLogin("Deadpool");
+        user.setAge(89);
+        user.setPassword("90809809");
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("Can`t register this user");
+        }
+    }
+
+    @Test
+    void loginHaveSizeLessThanAllowable_NotOk() {
+        user.setId(2l);
+        user.setLogin("sanya");
+        user.setPassword("7907001");
+        user.setAge(21);
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+            registrationService.register(user);
+                });
+    }
+
+    @Test
+    void loginHaveSizeEqualsAllowableSize_Ok() {
+        user.setId(4l);
+        user.setLogin("sanya2");
+        user.setPassword("7907001767");
+        user.setAge(23);
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("Can`t register this user");
+        }
+    }
+
+    @Test
+    void loginIsBiggerThanAllowableSize_Ok() {
+        user.setId(8l);
+        user.setLogin("sanya1256");
+        user.setPassword("7907001767");
+        user.setAge(23);
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("Can`t register this user");
+        }
+    }
+
+    @Test
+    void ageOfUserLessThanAllowableAge() {
+        user.setId(78l);
+        user.setAge(15);
+        user.setLogin("VenoM!!!");
+        user.setPassword("yufgugfyufguy");
+        Assertions.assertThrows(ImpossibleRegisterUserException.class,
+                ()-> {
+            registrationService.register(user);
+                });
+    }
+
+    @Test
+    void ageOfUserEqualsAllowableAge_Ok() {
+        user.setId(12l);
+        user.setAge(18);
+        user.setLogin("JaxTaller!");
+        user.setPassword("yufgugfyufguy1123");
+        try {
+            Assert.assertEquals(user, registrationService.register(user));
+        } catch (ImpossibleRegisterUserException e) {
+            System.out.println("Can`t register this user");
+        }
```

Comment:
Using `Assertions.assertDoesNotThrow` would be more appropriate here to ensure that the registration process completes without exceptions for valid inputs.